onnxruntime/core/providers/qnn/builder/opbuilder/gemm_op_builder.cc

```diff
@@ -92,7 +92,10 @@ Status GemmOpBuilder::ProcessInputs(QnnModelWrapper& qnn_model_wrapper,
     utils::InitializeQuantizeParam(quantize_param, is_quantized_tensor);
 
     const auto& input_name = inputs[input_i].node_arg.Name();
-    if (qnn_model_wrapper.IsQnnTensorWrapperExist(input_name)) {
+
+    // Only skip if the input tensor has already been added (by producer op) *and* we don't need
+    // to transpose it.
+    if (qnn_model_wrapper.IsQnnTensorWrapperExist(input_name) && input_trans_flag[input_i] == 0) {
```

Comment:
Note: This bug primarily prevented Gemm with dynamic activations (that are not graph inputs) from passing qnn validation.